### ISSUE TITLE
"This browser supports WebGPU but it appears to be disabled" on linux

### ISSUE THREAD

**Issue by joshspicer** [microsoft/vscode#235201]
Testing #234762

I thought i'd test on my Ubuntu 24.04 VM (virtualized on an ARM mac).  After enabling the setting and reloading I get this, presumably meaning my environment is incompatible?

![Image](https://github.com/user-attachments/assets/97baca51-0241-42c3-a32a-03ea8c6c34b1)

![Image](https://github.com/user-attachments/assets/a7f8dc95-5ae0-47eb-9b0d-e4525e076b7b)

When I click that button to revert back, I get this error (and nothing appears to happen behind the scenes):

![Image](https://github.com/user-attachments/assets/8c46cd4c-26fc-47e7-a27c-db2eef2a2d02)

**Comment by joshspicer** [comment]
I see your comment here https://github.com/microsoft/vscode/issues/235190#issuecomment-2515321896 so it sounds expected that I cannot test in a VM (although the error when hitting the button still seems like a bug).  Here's the VM's system info:

```
CPUs	unknown (10 x 0)
GPU Status	2d_canvas: unavailable_software canvas_oop_rasterization: disabled_off direct_rendering_display_compositor: disabled_off_ok gpu_compositing: disabled_software multiple_raster_threads: enabled_on opengl: disabled_off rasterization: disabled_software raw_draw: disabled_off_ok skia_graphite: disabled_off video_decode: disabled_software video_encode: disabled_software vulkan: disabled_off webgl: unavailable_software webgl2: unavailable_software webgpu: disabled_off webnn: unavailable_software
Load (avg)	3, 1, 1
Memory (System)	3.81GB (1.80GB free)
Process Argv	--crash-reporter-id 17306afd-079c-4956-8fb0-1cb216d6c7b5
Screen Reader	no
VM	0%
```

**Comment by Tyriar** [comment]
Yes it's expected, you may be able to enable GPU in your VM settings to workaround it but it's possible Chromium would ignore that.

**Comment by Tyriar** [comment]
Marked as feature as the wording could be improved and lead to docs. We don't handle switching without a reload gracefully yet.

**Comment by mubaidr** [comment]
Happens to me too. But VSCode shows that `webgpu` is enabled.

![Image](https://github.com/user-attachments/assets/084ebc3b-08fb-48c5-9e28-72ade67dbdbb)

```
Version: 1.97.0-insider
Commit: 63c0355e5bc04b124b8600ce42fe30c73b7a5a5e
Date: 2025-01-30T05:04:51.529Z (1 day ago)
Browser: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Code-Insiders/1.97.0-insider Chrome/128.0.6613.186 Electron/32.2.7 Safari/537.36
```

**Comment by Tyriar** [comment]
@mubaidr FYI this happens when `navigator.gpu.requestAdapter` returns undefined:

https://github.com/microsoft/vscode/blob/f6fb04a2bb19aa27b2f1d6a53ce45bbf2980bd80/src/vs/editor/browser/gpu/gpuDisposable.ts#L10-L26

@deepak1556 if webgpu is enabled as above, do you know why we wouldn't be returning an adapter here?

**Comment by mubaidr** [comment]
> @mubaidr FYI this happens when `navigator.gpu.requestAdapter` returns undefined:
> 
> https://github.com/microsoft/vscode/blob/f6fb04a2bb19aa27b2f1d6a53ce45bbf2980bd80/src/vs/editor/browser/gpu/gpuDisposable.ts#L10-L26
> 
> @deepak1556 if webgpu is enabled as above, do you know why we wouldn't be returning an adapter here?

@Tyriar understand. Also this is happening on both Linux and Windows. With two different gpu. I wonder why.

**Comment by bitfl0wer** [comment]
```
Name:                    gfx1031                            
  Uuid:                    GPU-XX                             
  Marketing Name:          AMD Radeon RX 6700 XT              
  Vendor Name:             AMD                                
  Feature:                 KERNEL_DISPATCH                    
  Profile:                 BASE_PROFILE                       
  Float Round Mode:        NEAR                               
  Max Queue Number:        128(0x80)                          
  Queue Min Size:          64(0x40)                           
  Queue Max Size:          131072(0x20000)                    
  Queue Type:              MULTI                              
  Node:                    1                                  
  Device Type:             GPU
```
```
Fedora Linux 41 (Workstation Edition) x86_64
```
I get the same error on Linux, using an AMD GPU. It has to be noted that I was using `codium` when this first happened, but a fresh (with my profile), official install from the Visual Studio Code website (.rpm package) yielded the same error message.

**Comment by Paladynee** [comment]
i'm using ubuntu 24 without a virtual machine and getting this error.

**Comment by clover256s** [comment]
@

> 我使用 ubuntu 24 没有虚拟机,并得到这个错误。 安装从 snap 商店

me to!!

**Comment by BladeMF** [comment]
I am using Ubuntu 24.10 with Wayland (dual boot) and I get "This browser supports WebGPU but it appears to be disabled: Error: This browser supports WebGPU but it appears to be disabled", but the VSCode window is on a second monitor rendered by my discrete Nvidia graphics card. This happens regardless of whether I am using "Launch with discrete graphics card or not".

**Comment by kibibites** [comment]
adding `--enable-unsafe-webgpu --enable-features=Vulkan,UseSkiaRenderer` to the launch argument resolved the issue for me
